src/SimpleSAML/Store/SQLStore.php

```diff
@@ -436,4 +448,25 @@ public function delete(string $type, string $key): void
         $query = $this->pdo->prepare($query);
         $query->execute($data);
     }
+
+
+    /**
+     * Calculates a base-64 encoded, URL-safe sha-256 hash.
+     *
+     * @param string $data
+     * @return string The hashed data.
+     */
+    public static function hashBase64(string $data) {
+        $hash = base64_encode(hash('sha256', $data, TRUE));
```

Comment:
May I suggest using the following instead?

```
$secretSalt = (new Utils\Config())->getSecretSalt();
return hash_hmac('sha256', $data, $secretSalt);
```

hash_hmac() will always produce a url-safe string.

Reply:
I also think this can be solved with a non-static private method, which is easier to unit-test.

Reply:
Using the salt will also provide more privacy

Reply:
Yes, that is a good idea.